clients/azuredevopsrepo/zip.go

```diff
@@ -0,0 +1,251 @@
+// Copyright 2024 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azuredevopsrepo
+
+import (
+	"archive/zip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7"
+
+	sce "github.com/ossf/scorecard/v5/errors"
+)
+
+const (
+	repoDir      = "repo*"
+	repoFilename = "azuredevopsrepo*.zip"
+	maxSize      = 100 * 1024 * 1024 // 100MB limit
+)
```

Comment:
Just to be clear, this is per file?

Reply:
Yes. I added this to resolve the G110 from gosec:

> G110: Potential DoS vulnerability via decompression bomb